src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultClaimUidExtractor.cs

```diff
@@ -38,6 +38,17 @@ public string ExtractClaimUid(ClaimsIdentity claimsIdentity)
         // Internal for testing
         internal static IEnumerable<string> GetUniqueIdentifierParameters(ClaimsIdentity claimsIdentity)
```

Comment:
We should only be looking at identities where the IsAuthenticated flag is true. If there are multiple ones, then either we choose the first which contains one of the claims we want, or perhaps look through all of them. I think using the first would be better, as we create "half" identities we throw away during social authentication.